demos/ret2spec_common.cc

```diff
@@ -0,0 +1,101 @@
+#include <array>
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+#include "cache_sidechannel.h"
+#include "instr.h"
+#include "ret2spec_common.h"
+#include "utils.h"
+
+// We cannot include the private_data twice, because that would lead to linking
+// error. Declaring it extern in here.
+extern const char *private_data;
+
+// Calls sched_yield in the cross-address-space version.
+void (*conditionally_unschedule)();
+// Calls sched_yield in the cross-address-space version, in the
+// same-address-space version starts the ReturnsFalse recursion.
+void (*unschedule_or_start_returns_false)();
+
+// Global variables used to avoid passing parameters through recursive function
+// calls. Since we flush whole stack frames from the cache, it is important not
+// to store on stack any data that might be affected by being flushed from
+// cache.
+size_t current_offset;
+const std::array<BigByte, 256> *oracle_ptr;
+
+// Return value of ReturnsFalse that never changes. Avoiding compiler
+// optimizations with it.
+bool false_value = false;
```

Comment:
This one might be a weird exception to the above, where the value has to be global to prevent the optimizer from removing it.